lib/src/main/java/com/okta/samples/Application.java

```diff
@@ -44,16 +28,49 @@
 
 @Controller
 @EnableAutoConfiguration
-public class Application {
+@SpringBootApplication
+@EnableOAuth2Sso
+public class Application extends WebSecurityConfigurerAdapter {
 
     // GLOBALS
     private OktaConfig CONFIG = getConfig();
-    private Map<String, Key> CACHED_KEYS = new HashMap<String, Key>();
     private User user = new User();
+    
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+      http
+        .antMatcher("/**")
+        .authorizeRequests()
+          .antMatchers("/", "/login**", "/webjars/**", "/assets/**","/authorization-code/login**")
+          .permitAll()
+        .anyRequest()
+          .authenticated();
+    }
+
+    @RequestMapping("/")
+    public String index(OAuth2Authentication authentication, Map<String, Object> model, HttpServletResponse response, HttpServletRequest request) {
+        if (authentication != null) {
+            Map<String, Object> details = (Map<String, Object>) authentication.getUserAuthentication().getDetails();
+            for (Object key : details.values()) {
+                System.out.println(key);    
+            }
+            user.setEmail(details.get("email").toString());
```

Comment:
FWIW, details.get() throws a NPE if the value doesn't exist for the user.

Reply:
Thanks @mraible! This is still a major WIP. Need to add a lot more checks before we can release it.
This PR is only a reference to use for @bdemers to use for upgrading our samples to v2. (use middleware libs and break-down front-end & back-end into separate repos)